core/src/smartcontracts/isi/account.rs

```diff
@@ -153,12 +158,16 @@ pub mod isi {
 
             wsv.modify_account(&account_id, |account| {
                 account.metadata_mut().insert_with_limits(
-                    self.key,
-                    self.value,
+                    self.key.clone(),
+                    self.value.clone(),
                     account_metadata_limits,
                 )?;
 
-                Ok(AccountEvent::MetadataInserted(account_id.clone()))
+                Ok(AccountEvent::MetadataInserted(MetadataChanged {
+                    origin_id: account_id.clone(),
```

Comment:
`origin_id` being the account whose metadata is being modified, or the account which modified the metadata? It's not obvious from the name, so I suggest you rename it.

Reply:
@appetrosyan i adopted naming used in `HasOrigin` trait, origin in this context mean account whose metadata is being modified. 
So `origin_id` is id of entity which is modified during isi execution.

Reply:
My comment stands. In this case a `target_account_id` is unambiguous, `origin_id` actually looks like the account that is doing the insertion. `HasOrigin`'s naming conventions can stay, because in that context, the origin makes sense (given that we're talking about events in general), while in this case, the naming should be clarified.

Reply:
I don't think that `target_accound_id` would fit because `MetadataInserted` is generic and used in domains, accounts, assets, ...
So maybe just `target_id`? What do you think?

Reply:
Decided to stick with `target_id`.